File: Extensions/Tessa.Extensions.Shared/Settings/Registrator.cs
using System;
using System.Collections.Generic;
using System.Text;
using Tessa.Platform.Settings;

namespace Tessa.Extensions.Shared.Settings
{
    [Registrator]
    public sealed class Registrator : RegistratorBase
    {
        public override void RegisterExtensions(IExtensionContainer extensionContainer)
        {
            extensionContainer
                .RegisterExtension<ISettingsExtension, PnrSettings>(x => x
                    .WithOrder(ExtensionStage.AfterPlatform, 1)
                    .WithSingleton())
                ;
        }
    }
}

File: WebClient SDK/src/PnrScripts/Views/registrator.ts
import { ExtensionContainer, ExtensionStage } from 'tessa/extensions';

import { PnrRefresh, PnrRefreshTreeItemExtension } from './PnrRefresh';

ExtensionContainer.instance.registerExtension({
  extension:PnrRefresh,
  stage: ExtensionStage.AfterPlatform,
  singleton: true
});

ExtensionContainer.instance.registerExtension({
  extension:PnrRefreshTreeItemExtension,
  stage: ExtensionStage.AfterPlatform,
  singleton: true
});